pkg/services/thumbs/service.go

```diff
@@ -284,3 +319,57 @@ func (hs *thumbService) getDashboardId(c *models.ReqContext, uid string) (int64,
 
 	return query.Result.Id, nil
 }
+
+func (hs *ThumbService) runOnDemandCrawl(parentCtx context.Context, theme models.Theme, mode CrawlerMode, kind models.ThumbnailKind, authOpts rendering.AuthOpts) {
+	crawlerCtx, cancel := context.WithTimeout(parentCtx, hs.scheduleOptions.maxCrawlDuration)
+	defer cancel()
+
+	// wait for at least a minute after the last completed run
+	interval := time.Minute
+	err := hs.lockService.LockAndExecute(crawlerCtx, hs.crawlLockServiceActionName, interval, func(ctx context.Context) {
+		hs.renderer.Run(crawlerCtx, authOpts, mode, theme, kind)
+	})
+
+	if err != nil {
+		tlog.Error("Error when on demand crawl", "err", err)
+	}
+}
+
+func (hs *ThumbService) runScheduledCrawl(parentCtx context.Context) {
+	crawlerCtx, cancel := context.WithTimeout(parentCtx, hs.scheduleOptions.maxCrawlDuration)
+	defer cancel()
+
+	err := hs.lockService.LockAndExecute(crawlerCtx, hs.crawlLockServiceActionName, hs.scheduleOptions.crawlInterval, func(ctx context.Context) {
+		hs.renderer.Run(crawlerCtx, rendering.AuthOpts{
+			OrgID:   0,
```

Comment:
`{ OrgId: 0, UserID: 0}` works when I run it locally, but I am sure there is a better solution